src/main/java/ParkingLot/AbstractVehicle.java

```diff
@@ -0,0 +1,163 @@
+package ParkingLot;
+
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static ParkingLot.VehicleParking.*;
+
+abstract public class AbstractVehicle {
```

Comment:
please reorder modifiers to 'public abstract class'

Reply:
done